### DISCUSSION TITLE
classC application of v4.4.5

### DISCUSSION THREAD

**Discussion by shabaz-git** [Lora-net/LoRaMac-node#1098] [Q&A] [answered]
Hi,
I am trying to run the classC application on B-L072Z-lrwan1 using 4.4.5 release(as latest release don't have a class C application). We are able to see the node joining and the very next uplink which is sending in class-c mode. We are trying to send the data from the chirpstack network server immediatley after on uplink (next uplink is after 60sec) such that the node should receive it in RX-continuous window.
`#define APP_TX_DUTYCYCLE                            (5000*12)`

 But in the Rx-continuos window we are not getting any log of receiving. The data is received during the next transmission in downlink-1 window.
![image](https://user-images.githubusercontent.com/70253206/118953827-cd831280-b97a-11eb-9274-5be78cf5d295.png)

The downlink queue is only getting cleared only after the next uplink.

Is this behaviour is expected as it looks like class A?

**Comment by mluis1** [comment, marked as answer]
The provided 4.4.5 periodic-uplink-lpp example can be used for ClassA, ClassB and ClassC operation.
By default the example is setup to operate in ClassA mode.

In order to operate in ClassC mode one must modify the `LORAWAN_DEFAULT_CLASS` definition to `CLASS_C` instead of `CLASS_A`.
https://github.com/Lora-net/LoRaMac-node/blob/2a983e16230a3b347237324969e3f40fceddbfba/src/apps/LoRaMac/periodic-uplink-lpp/B-L072Z-LRWAN1/main.c#L48-L51

From your description it looks like that the network server is not properly setup to handle the device in ClassC as you mention that the downlink is being performed in the ClassA Rx windows.

As this looks to be a network server configuration issue I would recommend that you look at [ChirpStack forum](https://forum.chirpstack.io/) if similar issue has been reported. If not please ask your question there.

**Reply by altishchenko** [reply]
It is a config issue - different servers have different means of specifying a downlink as a Class C, the server can't differentiate between A and C based on device's behaviour. The server we are using insists on setting application provided downlink with timestamp `immediate` or `imme`. Look around, there should be something like this in Chirpstack for sure.

**Reply by shabaz-git** [reply]
Thanks for the response. 
There is a class c support feature in the chirpstack.
![image](https://user-images.githubusercontent.com/70253206/119108098-785c0500-ba3d-11eb-9000-e0857f79a695.png)
After enabling this device is able to receive the downlink in Rx continuous window.

As this was saying a device can notify the network server about class with this mac command.
![image](https://user-images.githubusercontent.com/70253206/119107124-970dcc00-ba3c-11eb-81c1-8264e3036741.png)
Won't it be helpful if this mac command is also included in the stack?

**Reply by mluis1** [reply]
`DeviceModInd` is a MAC command introduced by LoRaWAN 1.1.x specifications. Only a 1.1.1 based stack will support this new MAC command.

LoRaMac-node v4.4.5 is based on LoRaWAN 1.0.3 specification. This means that this MAC command is not implemented.
